### PR TITLE
Change/lesson7

### PR DESCRIPTION
**Pull request by TetianaHarkusha**
Query Builder (with using  Blade)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,4 +13,5 @@ npm-debug.log
 yarn-error.log
 /.idea
 /.vscode
-.php-cs-fixer.cache
\ No newline at end of file
+.php-cs-fixer.cache
+storage/debugbar
```
Comment (fixmax) [inline .gitignore:17]: You can use already prepared gitignore: https://github.com/github/gitignore/blob/main/Laravel.gitignore

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -14,23 +15,123 @@ class UserController extends Controller
...
+                $users = DB::table('users')->where('age', '>', 30)->paginate(10);
+                break;
+            case 7://p.6 Users with age < 30
+                $users = DB::table('users')->where('age', '<', 30)->paginate(10);
+                break;
+            case 8://p.6 Users with age <= 30
+                $users = DB::table('users')->where('age', '<=', 30)->paginate(10);
+                break;
+            case 9://p.7 Users with age between 20 and 30
+                $columnNames = DB::getSchemaBuilder()->getColumnListing('users');
```
Comment (fixmax) [inline app/Http/Controllers/UserController.php:85]: Do nothing

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -14,23 +15,123 @@ class UserController extends Controller
...
+                $users = DB::table('users')->whereBetween('age', [20, 30])->paginate(10);
+                break;
+            case 10://p.8 Users with age=30 or salary=500 or id>4
+                $users = DB::table('users')->where('age', 30)->orWhere('salary', 500)->orWhere('id', '>', 4)->paginate(10);
+                break;
+            case 11://p.9 Users with age between 20 and 30 or salary between 400 and 800
+                $users = DB::table('users')->whereBetween('age', [20, 30])->orWhereBetween('salary', [400, 800])->paginate(10);
+                break;
+            case 12://p.10 Users without age between 30 and 40 order by 'salary' decrease
+                $users = DB::table('users')->whereNotBetween('age', [30, 40])->orderBy('salary', 'desc')->paginate(10);
```
Comment (fixmax) [inline app/Http/Controllers/UserController.php:95]: Optional, but better to use `->orderByDesc('salary')`

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -14,23 +15,123 @@ class UserController extends Controller
...
 
      /**
      * Show all users-admins
      */
     public function showAdminAll()
     {
-        return 'Список адміністраторів.';
+        return view('Users.newUser', [
+            'title' => 'admins-all',
+            'content' => file_get_contents(__DIR__ . '/../../../resources/Contents/adminsAll.html')
```
Comment (fixmax) [inline app/Http/Controllers/UserController.php:133]: Please don't use raw HTML.

File: database/factories/PostFactory.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+     *
+     * @return array<string, mixed>
+     */
+    public function definition()
+    {
+        $title = $this->faker->unique()->sentence();
+        $slug = str_replace(' ', '-', strtolower($title));
+        return [
+            'title' => $title,
+            'slug' => $slug,
```
Comment (fixmax) [inline database/factories/PostFactory.php:24]: Small logic/code can be in one line: `'title' => $this->faker->unique()->sentence(),`
And same for "slug"

File: database/migrations/2023_03_16_204408_create_posts_table.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('posts', function (Blueprint $table) {
+            $table->id();
+            $table->string('title', 100);
```
Comment (fixmax) [inline database/migrations/2023_03_16_204408_create_posts_table.php:17]: Maybe you need here field "user_id" for the HasMany relationship (User has many posts).

File: resources/Contents/adminsAll.html [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<h1>Список адміністраторів</h1>
+<p>Зараз ви можете переглянути повний список адміністраторів.</p> 
+<p>Більш детальна інформація надаєтьсмя за індивідуальним запитом.</p>
```
Comment (fixmax) [inline resources/Contents/adminsAll.html:3]: Do not use raw HTML, use blade or components instead

File: resources/views/Users/admin.blade.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>admin-{{$id}}</title>
+</head>
+<body>
+    <h1>Інформація про адміністратора:</h1>
+    <p>Ідентифікатор: {{$id}}</p>
+</body>
+</html>
```
Comment (fixmax) [inline resources/views/Users/admin.blade.php:13]: It should use the same logic as "layouts.app" (resources/views/Pages/user.blade.php)

File: resources/views/Users/adminsAll.blade.php [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>all-admins</title>
+</head>
+<body>
+    <h1>Список адміністраторів</h1>
+    <p>Зараз ви можете переглянути повний список адміністраторів.</p> 
+    <p>Більш детальна інформація надаєтьсмя за індивідуальним запитом.</p>
+</html>
```
Comment (fixmax) [inline resources/views/Users/adminsAll.blade.php:13]: It should use the same logic as "layouts.app" (resources/views/Pages/user.blade.php)
You can name it "layouts.admin"

File: resources/views/Users/user.blade.php [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>{{$title}}</title>
+</head>
+<body>
+    <h1>Інформація про користувача:</h1>
+    <p>Ідентифікатор: {{$id}}</p>
+    <p>Прізвище: {{$surname}}</p>
+    <p>Ім'я: {{$name}}</p>
+</body>
+</html>
```
Comment (fixmax) [inline resources/views/Users/user.blade.php:15]: Too use layout.

File: resources/views/Users/usersAll.blade.php [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>all-users</title>
+</head>
+<body>
+    <h1>Список користувачів</h1>
+    <p>Зараз ви можете переглянути повний список користувачів.</p> 
+    <p>Більш детальна інформація надаєтьсмя за індивідуальним запитом.</p>
+</body>
+</html>
```
Comment (fixmax) [inline resources/views/Users/usersAll.blade.php:14]: Too use layout.

File: routes/web.php [hunk 12]
```diff
@@ -18,9 +20,13 @@
 
 // routes for p.1
 Route::get('/', function () {
-    return 'Головна сторінка сайту';
+    return View::make('Pages.main', ['title' => 'StadyLaravel-main']);
 })->name('main');
 
+Route::get('/homework', function () {
+    return View::make('Pages.homework');
```
Comment (fixmax) [inline routes/web.php:27]: Here you can use just `view('Pages.homework')`.

File: routes/web.php [hunk 13]
```diff
@@ -72,3 +82,11 @@
...
     Route::get('/user/{id}', [UserController::class, 'showAdmin'])->name('user');
 });
+
+Route::get('/clear', function () {
+    Artisan::call('cache:clear');
+    Artisan::call('config:cache');
+    Artisan::call('view:clear');
+    Artisan::call('route:clear');
+    return "Кэш очищен.";
+});
```
Comment (fixmax) [inline routes/web.php:92]: For safe better to cover this code for only local use (prevent clearing cache on the production server).
isLocal() or app()->environment() == 'local'

Comment (TetianaHarkusha) [reply routes/web.php:92]: Thanks, i added this route for myself, but i'll take into account your remark

File: resources/views/Users/user.blade.php (hunk 10 above)
Comment (TetianaHarkusha) [reply resources/views/Users/user.blade.php:15]: i don't use this view, i'll delete it

**Pull request merged**